pkg/controller/job/job_controller.go

```diff
@@ -864,13 +897,37 @@ func getBackoff(queue workqueue.RateLimitingInterface, key interface{}) time.Dur
 	return calculated
 }
 
-// filterPods returns pods based on their phase.
-func filterPods(pods []*v1.Pod, phase v1.PodPhase) int {
+// countPodsByPhase returns pods based on their phase.
+func countPodsByPhase(pods []*v1.Pod, phase v1.PodPhase, job *batch.Job) int {
 	result := 0
-	for i := range pods {
-		if phase == pods[i].Status.Phase {
+	for _, p := range pods {
+		if phase == p.Status.Phase && (job.Spec.CompletionMode != batch.IndexedCompletion || getCompletionIndex(p.Annotations) != -1) {
 			result++
 		}
 	}
 	return result
 }
+
+func addSingleOrRangeStr(builder *strings.Builder, first, last int) {
```

Comment:
move this to the other file where it's used

Reply:
nice spot :)